### DISCUSSION TITLE
About FIN flag

### DISCUSSION THREAD

**Discussion by DHKim0428** [ANLAB-KAIST/KENSv3#70] [Project Q&A 2. Transport Layer (TCP)] [answered]
When we run the test-kens-close-run-solution and check the packet by Wireshark, there are no "FIN" flag sent / received.
Instead, it just sent / received (FIN | ACK). 
However, in the given TCP diagram, we need to send "FIN" then go to the state "FIN_WAIT_1" when close function is called. 
Is there any different close mechanism between the given diagram and the solution binary?

**Comment by Pusnow** [comment, marked as answer]
The TCP diagram may omit `ACK`.


https://github.com/ANLAB-KAIST/KENSv3/blob/2c57e617af4ace230271856f922de508a167e9f2/Documentation/rfc/rfc793.txt#L2484-L2503

In the RFC's closing sequence, the `ACK` flag is set.

**Reply by DHKim0428** [reply]
Thank you :)
